syntax = "proto3";

package Simulation.Data;

message EventData{
 //type ="FrameData";type="FreeSpaceRoadBound";type="TrafficLightInfo";
 string type=1;
 bytes payload=2;
}

message FrameData{
  int32 frameID = 1;
  //trackBoxs:动态障碍物
  repeated TrackBox trackBoxs= 2;
  //lans:车道线
  repeated Lane lans= 3;
  int32 speed = 4;
}

message FreeSpaceRoadBound{
  //boundData 道路区域 256*256*3二值化图像
  bytes boundData = 1;
}

message PointCloudV3Data{
  //pointClouds:点云
  repeated PointCloudV3 pointClouds= 1;
}

message TrafficLightInfo{
  //lightInfo:红绿灯
  repeated TrafficLight lightInfo = 1;
}

message VehicleMotion{
  //前轮转向角度
  float steerAngle = 1;
  //后轮驱动力
  float motorTorque = 2;
  //制动 默认0不制动
  float brakeTorque = 3;
}

message APAData{
   repeated ParkInfo apaParks = 1;
}

message ParkInfo{
  int32 id = 1;
  int32 number = 2;
  OccupyStatus status = 3;
  repeated PointV3 BoundingBoxs = 4;
  TypePark parkType = 5;

  enum OccupyStatus{
   //未知状态
   unknown = 0;
   //空闲状态
   free = 1;
   //占用状态
   Occupied = 2;
   //选择可入库状态
   destination = 3;
  }
  enum TypePark{
    unknownPark = 0;
	//垂直库位
    VerticalPark = 1;
	//水平库位
    HorizontalPark = 2;
  }
}

message PointV3{
 float x = 1;
 float y = 2;
 float z= 3;
}
message PointV2{
 float x = 1;
 float y = 2;
}
message PointCloudV3{
  PointV3 coord= 1;
  int32 type = 2;
}

message TrackBox{
  int32 objID= 1;
  //CarType
  int32 type= 2;
  //车辆坐标
  PointV3 centerPos= 3;
  //车辆朝向
  float heading= 4;
  float length= 5;
  float width= 6;
  float height= 7;
}

message Lane{
  // 1 虚车道线，2 行人斑马线，3 道路边界线，4 左右车道中间分界线（黄实线）
  int32 type = 1;
  repeated PointV2 points = 2;
}

message VehicleInfo{
  int32 speed = 1;
  float heading = 2;
  PointV3 VehicleWorldCoordinates= 3;
}

message TrafficLight{
  int32 status = 1;
  float x = 2;
  float y = 3;
  float z = 4;
}
enum CarType
{
    unknown = 0;
    bus = 1;
    car = 2;
    fiat = 3;
    furgao = 4;
    truck = 5;
}